sm_ipt/doc/usage.rst

```diff
@@ -0,0 +1,219 @@
+	}
+
+The above code uses the ``sm_ipt_alloc_tx_buf`` function to allocate transmit buffer in shared memory. Data is copied into said buffer and sent to another core via ``sm_ipt_send`` function.
+
+Data send no-copy
+-------------------
+
+The data is prepared after buffer allocation send to another core when data preparation (and usage) is finished.
+
+.. code-block:: c
+
+	int send_nocopy(size_t len)
+	{
+		uint8_t *buffer;
+		size_t i;
+
+		sm_ipt_alloc_tx_buf(&sm_ipt_ctx, &buffer, len);
+
+		/* Use buffer as any other allocated buffer */
+		some_data_preparation_function(buffer);
+
+		return sm_ipt_send(&sm_ipt_ctx, buffer, len);
+	}
+
+The above code uses the ``sm_ipt_alloc_tx_buf`` function to allocate transmit buffer in shared memory. Buffer is prepared by ``some_data_preparation_function`` and sent to another core via ``sm_ipt_send`` function.
+
+Data send abort
+-------------------
+
+There is possibility to free transmit buffer instead of sending it if the need arises.
+
+.. code-block:: c
+
+	int send_if(size_t len, bool send_choice)
+	{
+		uint8_t *buffer;
+		size_t i;
+
+		sm_ipt_alloc_tx_buf(&sm_ipt_ctx, &buffer, len);
+
+		if (send_choice){
+			/* Prepare some data */
+			some_data_preparation_function(buffer);
+
+			return sm_ipt_send(&sm_ipt_ctx, buffer, len);
+		}
+		else{
+			sm_ipt_free_tx_buf(&sm_ipt_ctx, buffer);
+
+			return ERROR_ABORT;
+		}
+	}
+
+The above code uses the ``sm_ipt_alloc_tx_buf`` function to allocate transmit buffer in shared memory. Buffer is send only if send_choice is true. If send_choice is false the buffer is free using ``sm_ipt_free_tx_buf`` and send operation is aborted.
+
+
+Data receive full-copy
+-------------------
+
+When data is received it can be processed using full-copy mechanism. Buffer has to be copied to local memory. Shared memory rx buffer has to be freed and data processing is done on local copy of buffer.
+
+.. code-block:: c
+
+	volatile uint8_t receive_buffer[MAX_PACKET_SIZE];
+	volatile bool data_received;
+
+	void sm_ipt_receive_handler(const uint8_t *packet, size_t len)
+	{
+		/* Copy whole message */
+		memcpy(receive_buffer, packet, len);
+
+		/* Free receive buffer */
+		sm_ipt_free_rx_buf(&sm_ipt_ctx, packet);
+
+		/* Signal message received */
+		data_received = true;
+	}
+
+	void data_processing_thread(void)
+	{
+		/* Wait for incoming data */
+		while(!data_received);
+
+		/*Process data */
+		some_data_processing_function(receive_buffer);
+	}
+
+The above code uses the ``memcpy`` function to copy shared memory buffer into local buffer. Next shared memory buffer is freed using ``sm_ipt_free_rx_buf``. Lastly the data is processed locally using a copy of buffer.
+
+
+Data receive no-copy
+-------------------
+
+When data is received it can be processed using no-copy mechanism. The pointer to shared memory buffer has to be saved in order to use it later. Shared memory rx buffer has to be freed and data processing is done on shared memory buffer.
```

Comment:
```suggestion
When the data is received, it can be processed using the no-copy mechanism.
The pointer to the shared memory buffer must be saved for later use.
The shared memory rx buffer must be freed and data processing takes place on the shared memory buffer.
```